### DISCUSSION TITLE
Is client private key required for mutual/two way ssl?

### DISCUSSION THREAD

**Discussion by KrishnaST** [Hakky54/ayza#74] [General]
My application is consuming a service which requires two way SSL. I have received public certificate(Most probably client certificate) from the vendor whose service i am going to consume. Is client private key required for mutual/two way ssl?

**Comment by Hakky54** [comment]
If the counter party, the service which you are going to call, requires you to authenticate with ssl based on mutual-ssl than you need two files:
- The certificate of the counter party, which you are going to use to mark the server as trusted server.
- Your keypair also known as private key and public key.

So indeed the private key is required. I have also here an example: [Integration test - mutual authentication with badssl.com](https://github.com/Hakky54/sslcontext-kickstart/blob/929269b0ff6351cbf0cdece76a78f0b17910e302/sslcontext-kickstart-for-pem/src/test/java/nl/altindag/ssl/SSLFactoryIT.java#L43)

If the private key is removed from the file which is being used by client within the test, it will fail. You can give it a try 😄
